### PR TITLE
feat(routing,supervision): update section about ValidateConfig

### PR DESCRIPTION
**Pull request by laplab**

### CODE REVIEW

File: src/ch04-02-supervision.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ What's the difference between default behavior and `with_force(true)` one? By de
 The reconfiguration process consists of several stages:
 ![](assets/reconfiguration.drawio.svg)
 
-1. Config validation: the configurer sends the `ValidateConfig` request to all groups and waits for responses.
-2. If all groups respond `Ok(_)` or ignore the request, the config is considered valid, and the configurer sends `UpdateConfig` to all groups. If at least one actor responds `Err(_)`, reconfiguration is aborted, and a new config isn't used.
+1. Config validation: the configurer sends the `ValidateConfig` request to all groups and waits for responses. If all groups respond `Ok(_)` or discard the request (which is the default behaviour), the config is considered valid and the configurer proceeds to the next step.
+2. Config update: the configurer sends `UpdateConfig` to all groups. If at least one actor responds `Err(_)`, reconfiguration is aborted, and a new config isn't used.
```
Comment (laplab) [inline src/ch04-02-supervision.md:26]: > and a new config isn't used

If I am understanding the code correctly, the new config will not be used only for the actor that returned `Err(_)`. Actors that returned `Ok(())` will use the new config. Is that the case? If yes, I think we should add clarification here.

File: src/ch04-01-routing.md [hunk 2]
```diff
@@ -117,7 +117,10 @@ Possible `Outcome`'s variants:
 * `Outcome::GentleMulticast` works like `Outcome::Multicast`, but doesn't lead to spawning new actors, instead a message is discarded.
 * `Outcome::Broadcast` sends to all active actors.
 * `Outcome::Discard` drops the message (that leads to an error on sending side).
-* `Outcome::Default` works as `Outcome::Discard` for user-defined messages, but as `Outcome::Broadcast` for the system ones, such as `UpdateConfig`, `ValidateConfig`, `Terminate` and so on.
+* `Outcome::Default` behaviour depends on the message type:
+    - `Outcome::Discard` for `ValidateConfig` message. Note that `ValidateConfig` message is designed to validate config against actor's state. Any other validation must happen on the deserialization stage.
```
Comment (loyd) [inline src/ch04-01-routing.md:121]: must → should

Comment (loyd) [reply src/ch04-01-routing.md:121]: Also note about the `validator` crate for complex validation (that can be run during deserialization)

Comment (loyd) [reply src/ch04-01-routing.md:121]: It's better to describe such things on a separate page and leave here only behavior notes.

File: src/ch04-02-supervision.md (hunk 1 above)
Comment (loyd) [reply src/ch04-02-supervision.md:26]: Nope, if any actor considers the config invalid, the next stage (`UpdateConfig`) is started at all. This provides consistency between configs if needed.

Comment (loyd) [reply src/ch04-02-supervision.md:26]: Ok, it's all about the second step. I think `UpdateConfig` can become just a regular message and by default `system.configurers` shouldn't wait until all actors have finished the update.

**Comment by laplab** [comment]
TODO: mention behaviour about tokens for requests sent as regular messages

**Pull request merged**